### ISSUE TITLE
Inconsistent multi-period weighting for the primary energy limit global constraint

### ISSUE THREAD

**Issue by coroa** [PyPSA/PyPSA#765]
## Checklist

- [x] I am using the current [`master`](https://github.com/pypsa/pypsa/tree/master) branch or the latest [release](https://github.com/pypsa/pypsa/releases). Please indicate:

## Describe the Bug

I just realized that the primary energy limit global constraints apply the number of years each period on-top of the regular generator snapshot weightings, but **only** for `generators`. And **not** for `storage_units`/`stores`.

ie. for generators the underlying assumption is that the dispatch repeats in each year of the investment period and therefore needs to be counted multiple times, while for storage units and stores only the difference between the last and the first level is counted without any weighting, which is inconsistent.

I had originally expected that the primary energy limit for a particular investment period would bound the **yearly** emissions, but i must admit the period weighting is necessary for an overall budget, then it is also necessary for a non-cyclic store/storage unit though.

@lisazeyen 

## Error Message

*If applicable, paste any terminal output to help illustrating your problem.*
*In some cases it may also be useful to share your list of installed packages: `conda list`.*

```
<paste here>
```

**Comment by fneum** [comment]
Yes, we should have resolved this almost two years ago! Will look into it soon.
